Add support for an operator level thread local observer

rohithmenon opened the pull request:
Summary: Add support for an operator level thread local observer

Test Plan: Verified the interception as part of a pytorch model evaluation with static runtime.

Reviewed By: davidberard98

Differential Revision: D49025762

rohithmenon commented:
Re-triggering easycla check  /easycla

davidberard98 reviewed (approved):
LGTM pending tests passing

rohithmenon commented:
/easycla

Commit:
Add support for an operator level thread local observer (#108816)

Summary:

Add support for an operator level thread local observer

Test Plan: Verified the interception as part of a pytorch model evaluation with static runtime.

Reviewed By: davidberard98

Differential Revision: D49025762
